src/Mod/PartDesign/App/FeatureSketchBased.cpp

```diff
@@ -666,6 +705,154 @@ void ProfileBased::generatePrism(TopoDS_Shape& prism,
+                                        const double angle2,
+                                        std::list<TopoDS_Shape>& drafts)
+{
+    // inspect sketchshape
+    // that it is not null was already checked
+    TopoDS_Wire sourceWire;
+    if (sketchshape.ShapeType() == TopAbs_WIRE) {
+        ShapeFix_Wire aFix;
+        aFix.Load(TopoDS::Wire(sketchshape));
+        aFix.FixReorder();
+        aFix.FixConnected();
+        aFix.FixClosed();
+        sourceWire = aFix.Wire();
+    }
+    else if (sketchshape.ShapeType() == TopAbs_FACE) {
+        TopoDS_Wire outerWire = ShapeAnalysis::OuterWire(TopoDS::Face(sketchshape));
+        sourceWire = outerWire;
+    }
+    else if (sketchshape.ShapeType() == TopAbs_COMPOUND) {
+        TopoDS_Iterator it(sketchshape);
+        for (; it.More(); it.Next()) {
+            generateTaperedPrism(it.Value(), direction, Length, Length2, angle, angle2, drafts);
+        }
+    }
+    else {
+        throw Base::RuntimeError("Only a wire or a face is supported");
+    }
+
+    if (sourceWire.IsNull()) {
+        throw Base::RuntimeError("Offset for base shape could not made");
+    }
+
+    // calculate lateral offset and vectors for top and bottom shape of the loft
+    double lateralOffset = tan(angle * M_PI / 180.0) * Length;
+    double lateralOffset2 = tan(angle2 * M_PI / 180.0) * Length2;
+    gp_Vec vecOffset = gp_Vec(direction) * Length;
+    gp_Vec vecOffset2 = gp_Vec(direction.Reversed()) * Length2;
+
+    // create a new shape which is the from shape with the offset defined by the TaperAngle
+    // at first inspect the from shape
+    int numEdges = 0;
+    TopExp_Explorer xp(sourceWire, TopAbs_EDGE);
+    while (xp.More()) {
+        numEdges++;
+        xp.Next();
+    }
+    // Pad / Pocket does not only support closed wires, thus only 1 edge is not valid
+    // if this might be supported in future, see Extrusion::makeDraft how to handle 1 edge
+    if (numEdges == 1)
+        throw Base::RuntimeError("Base shape consists of only one edge");
+
+    // to store the sections for the loft
+    std::list<TopoDS_Wire> list_of_sections;
+    BRepBuilderAPI_MakeWire sectionResult;
+    
+    // if we have a second direction length create a bottom section
+    if (Length2 > Precision::Confusion()) {
+        // this offsetWire must be the first loft section
+        createTaperedPrismOffset(sourceWire, vecOffset2, lateralOffset2, Length2, true, sectionResult);
+        list_of_sections.push_back(sectionResult);
+    }
+
+    // since the order is important store the other sections after bottom
+    list_of_sections.push_back(sourceWire);
+    // create the top section
+    createTaperedPrismOffset(sourceWire, vecOffset, lateralOffset, Length, false, sectionResult);
+    list_of_sections.push_back(sectionResult);
+    
+    // build the loft using the created sections
+    BRepOffsetAPI_ThruSections mkGenerator(Standard_True, /*ruled=*/Standard_True);

```

Comment:
Without precompiled headers, this requires an `# include <BRepOffsetAPI_ThruSections.hxx>` here on apple-clang-11.0

Reply:
thanks, added